### PR TITLE
Add predict from list of images from text file

### PR DESCRIPTION
**Pull request by jsalbert**
- This PR adds predict from list of image paths
- Also fixes the possibility of adding verbose to predict function

### CODE REVIEW

File: keras_eval/eval.py [hunk 1]
```diff
@@ -337,14 +339,39 @@ def _predict_folder(self, folder_path, verbose):
...
 
         self.probabilities = np.array(probabilities)
         self.combined_probabilities = utils.combine_probabilities(self.probabilities, self.combination_mode)
         self.image_paths = image_paths
 
         return self.probabilities
 
+    def _predict_list(self, image_list, verbose=True):
+        '''
+
```
Comment (adriaromero) [inline keras_eval/eval.py]:
```suggestion
```

File: keras_eval/eval.py [hunk 2]
```diff
@@ -337,14 +339,39 @@ def _predict_folder(self, folder_path, verbose):
...
+        Predict the class probabilities of a set of images from a list.
+
+        Args:
+            image_list: List of images
+
+        Returns: Probabilities predicted, image path for every image (aligned with probability)
+
```
Comment (adriaromero) [inline keras_eval/eval.py]:
```suggestion
```

File: keras_eval/eval.py [hunk 3]
```diff
@@ -337,14 +339,39 @@ def _predict_folder(self, folder_path, verbose):
...
 
         return self.probabilities
 
+    def _predict_list(self, image_list, verbose=True):
+        '''
+
+        Predict the class probabilities of a set of images from a list.
+
+        Args:
+            image_list: List of images
```
Comment (adriaromero) [inline keras_eval/eval.py]: Am I right?
```suggestion
            image_list: List of image paths.
```

File: keras_eval/eval.py [hunk 4]
```diff
@@ -337,14 +339,39 @@ def _predict_folder(self, folder_path, verbose):
...
+
+        Returns: Probabilities predicted, image path for every image (aligned with probability)
+
+        '''
+        probabilities = []
+        image_paths = []
+
+        for image_path in image_list:
```
Comment (adriaromero) [inline keras_eval/eval.py]: Should we first check if the `image_list` is correct? i.e. checking that files exists

File: keras_eval/eval.py [hunk 5]
```diff
@@ -337,14 +339,39 @@ def _predict_folder(self, folder_path, verbose):
...
+            # Read images from folder
+            if image_path.endswith(".png") or image_path.endswith(".jpeg") or image_path.endswith(".jpg"):
+                probabilities.append(self._predict_image(image_path, verbose)[0])
+                image_paths.append(image_path)
+
+        self.probabilities = np.array(probabilities)
+        self.image_paths = image_paths
+
+        return self.probabilities
```
Comment (adriaromero) [inline keras_eval/eval.py]: Are you saying in the function description that it returns the `image_paths` as well?

File: tests/conftest.py [hunk 6]
```diff
@@ -32,6 +32,17 @@ def test_image_path():
...
 
 
+@pytest.fixture('session')
+def test_image_paths_list():
+    return [os.path.abspath(os.path.join('tests', 'files', 'catdog', 'test', 'cat', 'cat-1.jpg')),
+            os.path.abspath(os.path.join('tests', 'files', 'catdog', 'test', 'dog', 'dog-2.jpg'))]
+
+
+@pytest.fixture('session')
+def test_ensemble_models_path():
```
Comment (adriaromero) [inline tests/conftest.py]: Do you still need this fixture?

Comment (jsalbert) [reply tests/conftest.py]: I didn't know it wasn't used anymore

**Comment by jsalbert** [comment]
Ok, made the changes requested!  🚀 🎸

File: keras_eval/eval.py (hunk 5 above)
Comment (jsalbert) [reply keras_eval/eval.py]: It was just copied from the predict function, which was wrong too, changed it

**Pull request merged**